### PR TITLE
0.11.0 packager integration

### PR DESCRIPTION
**Pull request by igorshevach**

### CODE REVIEW

File: common/config/config.json.template [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
```
Comment (kalturaguy) [inline common/config/config.json.template]: revert back to original

File: lib/entry/FlavorDownloader.js [hunk 2]
```diff
@@ -145,30 +145,42 @@ var FlavorDownloader = (function() {
...
                 }
-                return Q.resolve();
-        });
+                return Q.resolve(_.map(results,function(p){
+                    return p.value;
+                }));
+            });
+    };
+
+    var removeFile = function (pathToDelete) {
```
Comment (kalturaguy) [inline lib/entry/FlavorDownloader.js]: shouldn't it be in playlistGenerator?

File: lib/entry/FlavorDownloader.js [hunk 3]
```diff
@@ -145,30 +145,42 @@ var FlavorDownloader = (function() {
...
+        var that = this;
+
+        return qfs.remove(pathToDelete)
+            .then(function () {
+                that.logger.info('Successfully deleted chunk ' + path.basename(pathToDelete));
```
Comment (kalturaguy) [inline lib/entry/FlavorDownloader.js]: use %s instead of +

File: lib/entry/FlavorDownloader.js [hunk 4]
```diff
@@ -202,43 +218,41 @@ var FlavorDownloader = (function() {
...
-                });
+                    });
             }
 
             return newChunks;
         };
+
+
+
         var processObsoleteChunks = function(results) {
```
Comment (kalturaguy) [inline lib/entry/FlavorDownloader.js]: playlistGenerator?

File: lib/manifest/PlaylistGenerator.js [hunk 5]
```diff
@@ -1,26 +1,340 @@
...
+                that.playlistImp.initialSegmentIndex = Math.floor((that.playlistImp.firstClipTime - that.playlistImp.segmentBaseTime) / segmentDuration);
+            }
+
+        } else {
+            delete that.minMax;
+        }
+        that.isDirty = false;
+        try {
+            var str = JSON.stringify(that.playlistImp);
+            return that.persistPromise = fsUtils.writeFileAtomically(that.playlistPath,JSON.stringify(that.playlistImp));
```
Comment (kalturaguy) [inline lib/manifest/PlaylistGenerator.js]: remove double stringify

File: lib/manifest/PlaylistGenerator.js [hunk 6]
```diff
@@ -1,26 +1,340 @@
...
+        } catch(err){
+            return Q.resolve(err);
+        }
+    }
+    return qResolved;
+};
+
+var finaizeStart = function (){
```
Comment (kalturaguy) [inline lib/manifest/PlaylistGenerator.js]: mispelling

File: lib/manifest/PlaylistGenerator.js [hunk 7]
```diff
@@ -1,26 +1,340 @@
...
+
+PlaylistGenerator.prototype.start = function start() {
+    var that = this;
+
+    if(that.startPromise)
+        return that.startPromise;
+
+    try {
+        that.startPromise = Q.defer();
+        fs.exists(that.playlistPath,function(ok) {
```
Comment (kalturaguy) [inline lib/manifest/PlaylistGenerator.js]: rather if you use promises here, it's one time thing, and it's better to be consist

File: lib/utils/playlistGenrator-utils.js [hunk 8]
```diff
@@ -0,0 +1,1104 @@
...
+        seq.clips = seq.clips.map( function(c) {
+            return unserializeClip(c);
+        });
+    });
+};
+
+Playlist.prototype.addListener = function(listener){
+    var that = this;
+
+    BroadcastEventEmitter.prototype.addListener.call(this,listener);
```
Comment (kalturaguy) [inline lib/utils/playlistGenrator-utils.js]: you should derive from EventEmitter and remove the addListenr and removeListener

File: lib/utils/playlistGenrator-utils.js [hunk 9]
```diff
@@ -0,0 +1,1104 @@
+/**
```
Comment (kalturaguy) [inline lib/utils/playlistGenrator-utils.js]: I rather if you create a folder under /lib/playlistGenerator  and not utils
+ each class in different file

File: lib/utils/playlistGenrator-utils.js (hunk 8 above)
Comment (igorshevach) [inline lib/utils/playlistGenrator-utils.js]: EventEmitter is very slow

File: lib/Driver.js [hunk 10]
```diff
@@ -12,7 +12,7 @@ var os = require('os');
 http.globalAgent.maxSockets = Infinity;
 
 if (os.platform() === 'darwin') {
-    http.globalAgent.maxSockets = 10;
+    http.globalAgent.maxSockets = 50;
```
Comment (kalturaguy) [inline lib/Driver.js]: was this needeD?

File: lib/Worker.js [hunk 11]
```diff
@@ -110,7 +110,7 @@ module.exports = (function(){
         });
         _.forEach(alreadyLiveEntries, function(e) {
             var entry = that.handledEntries[e.entryId];
-            if (entry !== 'pending')
+            if (entry !== 'pending' && entry !== 'cancelPending')
```
Comment (kalturaguy) [inline lib/Worker.js]: @gadyaari  Gad, can you please talk with Igor about the cancelPending to see if this is missing from 11.0 as well

**Pull request closed without merging**